plinth/modules/dynamicdns/templates/dynamicdns_configure.html

```diff
@@ -0,0 +1,168 @@
+              $('#id_dynamicdns-use_http_basic_auth').prop('checked', true);
+            }else{
+              $('#id_dynamicdns-use_http_basic_auth').prop('checked', false);
+            }
+            if ($("#id_dynamicdns-service_type option:selected").text() == "selfhost.bz") {
+              $('#id_dynamicdns-dynamicdns_update_url').val(SELFHOST);
+            }
+            if ($("#id_dynamicdns-service_type option:selected").text() == "freedns.afraid.org") {
+                $('#id_dynamicdns-dynamicdns_update_url').val(FREEDNS);
+              }
+            if ($("#id_dynamicdns-service_type option:selected").text() == "other update URL") {
+              $('#id_dynamicdns-dynamicdns_update_url').val('');
+            }
+          }
+      });
+
+      $('#id_dynamicdns-showpw').change(function() {
+        //changing type attribute from password to text is prevented by
+        //most browsers make a new form field works for me
+        if ($('#id_dynamicdns-showpw').prop('checked')) {
+              $('#id_dynamicdns-dynamicdns_secret').replaceWith(
+                      $('#id_dynamicdns-dynamicdns_secret').clone().attr(
+                              'type', 'text'));
+        }else{
+              $('#id_dynamicdns-dynamicdns_secret').replaceWith(
+                      $('#id_dynamicdns-dynamicdns_secret').clone().attr(
+                              'type', 'password'));
+        }
+      });
+
+      function select_service()
+      {
+        if ( $("#id_dynamicdns-dynamicdns_server").val().length == 0 ) {
+          set_update_url_mode()
+          if($("#id_dynamicdns-dynamicdns_update_url").val() == NOIP){
+            $("#id_dynamicdns-service_type").val(2);
+          }else if($("#id_dynamicdns-dynamicdns_update_url").val() == SELFHOST){
+            $("#id_dynamicdns-service_type").val(3);
+          }else if ($("#id_dynamicdns-dynamicdns_update_url").val() == FREEDNS){
+            $("#id_dynamicdns-service_type").val(4);
+          }else{
+            $("#id_dynamicdns-service_type").val(5);
+          }
+        }else{
+          $("#id_dynamicdns-service_type").val(1);
+          set_gnudip_mode();
+        }
+      }
+
+      function set_gnudip_mode()
+      {
+        $('#id_dynamicdns-dynamicdns_update_url').closest('.form-group').hide();
+        $('#id_dynamicdns-disable_SSL_cert_check').closest('.form-group').hide();
+        $('#id_dynamicdns-use_http_basic_auth').closest('.form-group').hide();
+        $('#id_dynamicdns-dynamicdns_server').closest('.form-group').show();
+      }
+
+      function set_update_url_mode()
+      {
+        $('#id_dynamicdns-dynamicdns_update_url').closest('.form-group').show();
+        $('#id_dynamicdns-disable_SSL_cert_check').closest('.form-group').show();
+        $('#id_dynamicdns-use_http_basic_auth').closest('.form-group').show();
+        $('#id_dynamicdns-dynamicdns_server').closest('.form-group').hide();
+      }
+
+      function show_all()
+      {
```

Comment:
instead of the lines here it'd be better to use `$('form.form .form-group').show();`
add a custom class to the form (i.e. `dynamicdns`) to make it a bit safer, then you can use `$('form.dynamicdns .form-group').show();`